### ISSUE TITLE
[Feature Request] Custom Behavior on `NUL`-byte

### ISSUE THREAD

**Issue by sisungo** [max-heller/tracing-syslog#3]
## Introduction
Currently, this crate `panic!`'s in dev builds, and `eprintln!`'s an error message in release builds when `NUL`-byte was found in the log string. This issue requests a new feature, which allows the user to customize the default behavior.

## Example
For example, the user may want to replace all `\0` to `U+FFFD` or directly remove `\0`'s.

**Comment by max-heller** [comment]
Thanks for the feature request!

Would something like the following work for your use case?

```rust
impl Syslog {
    pub fn invalid_chars(self, cfg: InvalidChars) -> Self { ... }
}
pub enum InvalidChars {
    Discard, // remove invalid chars from string
    SkipEvent, // current behavior in release mode
    Panic, // current behavior in dev mode
    ReplaceWith(char), // replace invalid chars with e.g. `U+FFFD`
}
```

Or do you anticipate needing something more generic like the ability to pass an arbitrary function pointer?

**Comment by sisungo** [comment]
Yes, something like this works with my use case.

**Comment by sisungo** [comment]
Yes, something like this works with my use case.



> Thanks for the feature request!
> 
> Would something like the following work for your use case?
> 
> ```rust
> impl Syslog {
>     pub fn invalid_chars(self, cfg: InvalidChars) -> Self { ... }
> }
> pub enum InvalidChars {
>     Discard, // remove invalid chars from string
>     SkipEvent, // current behavior in release mode
>     Panic, // current behavior in dev mode
>     ReplaceWith(char), // replace invalid chars with e.g. `U+FFFD`
> }
> ```
> 
> Or do you anticipate needing something more generic like the ability to pass an arbitrary function pointer?

**Comment by max-heller** [comment]
Released as part of [v0.3.0](https://crates.io/crates/syslog-tracing/0.3.0), sorry for the long wait
